contrib/pmsnare/pmsnare.c

```diff
@@ -81,11 +84,10 @@ CODESTARTisCompatibleWithFeature
 		iRet = RS_RET_OK;
 ENDisCompatibleWithFeature
 
-
 BEGINparse
 	uchar *p2parse;
 	int lenMsg;
-	int snaremessage;
+	int len_snare_tag;
 	int tablength;
 
 CODESTARTparse
```

Comment:
The entire reason for having TabRepresentation and tablength was to try and provide a clean way of handling tabs that are represented in different ways.

The "normal" case is tabs escaped (because we should be escaping control characters and there isn't a way to escape all control characters except for tab)

The idea was that by changing TabRepresentation to match the escaping (or lack of escaping), the rest of the code would not need to have tests for the various possible representations.

you should be able to just change TabRepresentation to "\t" and have it match unescaped tabs and 'do the right thing'

I never went back to actually implement and test it.

If you are not going to make use of this (and instead add separate tests for \t vs #011) then you should simplify the code and rip out all the tabrepresentation and tablength stuff.

But I think this entire commit could be replaced by some code that detected if tabs are escaped or not (and if they are escaped into octal or decimal) and sets tabrepresentation appropriately